pkg/abstractions/image/build.go

```diff
@@ -387,3 +367,28 @@ func (b *Builder) generatePipInstallCommand(opts *BuildOpts) string {
 	packages := strings.Join(escapedPackages, " ")
 	return fmt.Sprintf("%s -m pip install --root-user-action=ignore %s", opts.PythonVersion, packages)
 }
+
+var imageNamePattern = regexp.MustCompile(`^(?:([^/]+)(?:/))?((?:[^/:]+/)?[^/:]+)(?::([^/]+))?$`)
```

Comment:
Is there a source for this? We should probably add a comment to some url where this verified

Reply:
There isn't. It was trial and error. There are tests for this that should cover all cases if there's a concern.